core/domain/exp_services.py

```diff
@@ -1344,6 +1351,11 @@ def revert_exploration(
     # not add the committer of the revert to the list of contributors.
     update_exploration_summary(exploration_id, None)
 
+    # Save state id mapping model for the new exploration version.
+    if feconf.ENABLE_STATE_ID_MAPPING:
```

Comment:
@seanlip I introduced this to fix backend tests because many of the backend tests were making direct modifications to exploration domain object without any consent of change_list. There are many failing tests and I am still analyzing the log file.
Tests have failed because of storing same exploration version twice, making direct changes to exploration without making any changes to change_list, the previous version of exploration was stored without the state_id_mapping model for it. What I want to know is whether such situations can arise in production?

Reply:
I don't believe so, but I recommending adding asserts to your job to verify that. I think production is clean with respect to this, but it's worth double-checking.

Reply:
What sort of asserts are you suggesting?

Reply:
Actually, on reflection, the only test I can think of is to replay the save (i.e. verify that old exploration + change list == new exploration). The other cases won't happen. But on second thoughts that's a bit overkill and I think you can assume that that's not something that will happen in production. If we wanted to be paranoid, we could write a one-off validation job that iterates through all explorations and checks just that.